### PR TITLE
Enhancement: Run psalm using GitHub actions

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] runs `vimeo/psalm` (as installed with `phive` to `/tools`) using GitHub actions

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,3 +18,20 @@ jobs:
...
+
+      - name: Update dependencies with composer
+        uses: docker://php:7.3-cli
+        with:
+          args: composer update
+
+      - name: Run vimeo/psalm
+        uses: docker://php:7.3-cli
+        with:
+          args: ./tools/psalm --config=.psalm/static-analysis.xml --no-progress --show-info=false && ./tools/psalm --config=.psalm/config.xml --no-progress --shepherd --stats --show-info=false
```
Comment (localheinz) [inline .github/workflows/ci.yml]: Not sure, I need to investigate if there's a better way to run multiple commands - perhaps as a separate step?

File: .travis.yml [hunk 2]
```diff
@@ -50,16 +50,6 @@ notifications:
...
 jobs:
   include:
-    - stage: "Static Code Analysis"
-      php: 7.3
-      env:
-        - TOOL="psalm"
-      install:
-        - phpenv config-rm xdebug.ini
-      script:
-        - travis_retry ./tools/composer update $DEFAULT_COMPOSER_FLAGS
```
Comment (localheinz) [inline .travis.yml]: Note to self: need to specify flags.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -18,3 +18,25 @@ jobs:
...
+
+  type-checker:
+    name: Type Checker
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@master
+
+      - name: Update dependencies with composer
+        uses: docker://composer:1.9.0
```
Comment (Slamdunk) [inline .github/workflows/ci.yml]: Composer docker image can have a PHP version different from the targeted one of the following step, so there can be a mismatch of the installed dependencies, right?

**Comment by Ocramius** [comment]
For now, this would suffice:

```
./tools/psalm --config=.psalm/config.xml --set-baseline=.psalm/baseline.xml
```

Please be sure to review the diff on `baseline.xml` before merging.

**Review by Ocramius — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (localheinz) [reply .github/workflows/ci.yml]: Yes, that's definitely a problem we need to solve by providing dedicated Docker images!

**Comment by localheinz** [comment]
Thank you, @sebastianbergmann!

**Pull request merged**